hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterTrashMultipleDestinationMountTableResolver.java

```diff
@@ -0,0 +1,196 @@
+  private static final String FILE_NS1 = MOUNT_POINT_CHILD_DIR + "/fileNs1";
+  private static final String TRASH_ROOT = "/user/" + TEST_USER + "/.Trash";
+  private static final String CURRENT = "/Current";
+
+  @BeforeClass
+  public static void globalSetUp() throws Exception {
+    // Build and start a federated cluster
+    cluster = new StateStoreDFSCluster(false, 2,
+        MultipleDestinationMountTableResolver.class);
+    Configuration routerConf =
+        new RouterConfigBuilder().stateStore().admin().quota().rpc().build();
+
+    Configuration hdfsConf = new Configuration(false);
+    hdfsConf.setBoolean(DFSConfigKeys.DFS_NAMENODE_ACLS_ENABLED_KEY, true);
+    hdfsConf.set("fs.trash.interval", "1440");
+    hdfsConf.set("fs.trash.checkpoint.interval", "1440");
+    cluster.addRouterOverrides(routerConf);
+    cluster.addNamenodeOverrides(hdfsConf);
+    cluster.startCluster();
+    cluster.startRouters();
+    cluster.waitClusterUp();
+
+    ns0 = cluster.getNameservices().get(0);
+    ns1 = cluster.getNameservices().get(1);
+
+    nnContextNs0 = cluster.getNamenode(ns0, null);
+    nnFsNs0 = nnContextNs0.getFileSystem();
+    nnContextNs1 = cluster.getNamenode(ns1, null);
+    nnFsNs1 = nnContextNs1.getFileSystem();
+
+    routerContext = cluster.getRandomRouter();
+    resolver =
+        (MultipleDestinationMountTableResolver) routerContext.getRouter().getSubclusterResolver();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    if (cluster != null) {
+      cluster.stopRouter(routerContext);
+      cluster.shutdown();
+      cluster = null;
+    }
+  }
+
+  private boolean addMountTable(final MountTable entry) throws IOException {
+    RouterClient client = routerContext.getAdminClient();
+    MountTableManager mountTableManager = client.getMountTableManager();
+    AddMountTableEntryRequest addRequest =
+        AddMountTableEntryRequest.newInstance(entry);
+    AddMountTableEntryResponse addResponse =
+        mountTableManager.addMountTableEntry(addRequest);
+    // Reload the Router cache
+    resolver.loadCache(true);
+    return addResponse.getStatus();
+  }
+
+  @Test
+  public void testMoveToTrashWithMultipleDestinationMountTableResolver() throws IOException,
+      URISyntaxException, InterruptedException {
+
+    // add MountPoint  /home/data  ns0 -> /home/data, ns1 -> /home/data
+    Map<String, String> destMap = new HashMap<>();
+    destMap.put(ns0, MOUNT_POINT);
+    destMap.put(ns1, MOUNT_POINT);
+    MountTable addEntry = MountTable.newInstance(MOUNT_POINT, destMap);
+    addEntry.setDestOrder(DestinationOrder.HASH_ALL);
+    assertTrue(addMountTable(addEntry));
+
+    // current user client ,supper user setup permission for testUser
+    DFSClient clientNs0 = nnContextNs0.getClient();
+    DFSClient clientNs1 = nnContextNs1.getClient();
+
+    clientNs0.setOwner("/", TEST_USER, TEST_USER);
+    clientNs1.setOwner("/", TEST_USER, TEST_USER);
+
+    UserGroupInformation ugi = UserGroupInformation.
```

Comment:
Single line

Reply:
OK , I will submit it together after we confirm the above changes.